StoneMasonKarel.java

```diff
@@ -1,17 +1,85 @@
-/*
- * File: StoneMasonKarel.java
- * --------------------------
- * The StoneMasonKarel subclass as it appears here does nothing.
- * When you finish writing it, it should solve the "repair the quad"
- * problem from Assignment 1.  In addition to editing the program,
- * you should be sure to edit this comment so that it no longer
- * indicates that the program does nothing.
- */
-
-import stanford.karel.*;
-
-public class StoneMasonKarel extends SuperKarel {
-
-	// You fill in this part
-
-}
+/*
+ * File: StoneMasonKarel.java
+ * --------------------------
+ * The StoneMasonKarel subclass as it appears here does nothing.
+ * When you finish writing it, it should solve the "repair the quad"
+ * problem from Assignment 1.  In addition to editing the program,
+ * you should be sure to edit this comment so that it no longer
+ * indicates that the program does nothing.
+ */
+
+import stanford.karel.*;
+
+public class StoneMasonKarel extends SuperKarel {
+
+	 public void run() {
+		 turnLeft();
+		 moveOn();
+		 moveOn();
+		 moveOn();
+		 move();
+		 turnRight();
+		 moveFour();
+		 turnRight();
+		 moveOn();
+		 move();
+		 moveOn();
+		 move();
+		 
+		 turnLeft();
+		 moveFour();
+		 turnLeft();
+		 moveOn();
+		 moveOn();
+		 move();
+		 moveOn();
+		 turnRight();
+		 moveFour();
+		 
+		 turnRight();
+		 moveOn();
+		 moveOn();
+	 }
+	 
+	 private void moveOn(){

```

Comment:
I think you could have picked a better method name for this function. The code within it seems to be a lot and it doesn't seem like "moveOn:" describe it well. Also in the future comment your methods to help explain what is happening.